client/my-sites/plans-features-main/index.tsx

```diff
@@ -476,39 +454,24 @@ const PlansFeaturesMain = ( {
 	} );
 
 	// we neeed only the visible ones for comparison grid (these should extend into plans-ui data store selectors)
-	const gridPlansForComparisonGrid = useMemo( () => {
-		const hiddenPlans = [ PLAN_HOSTING_TRIAL_MONTHLY, PLAN_ENTERPRISE_GRID_WPCOM ];
-
-		return filteredPlansForPlanFeatures.reduce( ( acc, gridPlan ) => {
-			if ( gridPlan.isVisible && ! hiddenPlans.includes( gridPlan.planSlug ) ) {
-				return [
-					...acc,
-					{
-						...gridPlan,
-						features: planFeaturesForComparisonGrid[ gridPlan.planSlug ],
-					},
-				];
-			}
-
-			return acc;
-		}, [] as GridPlan[] );
-	}, [ filteredPlansForPlanFeatures, planFeaturesForComparisonGrid ] );
+	const gridPlansForComparisonGrid = useGridPlansForComparisonGrid( {
+		allFeaturesList: FEATURES_LIST,
+		gridPlans: filteredPlansForPlanFeatures,
+		intent,
+		selectedFeature,
+		showLegacyStorageFeature,
+	} );
 
 	// we neeed only the visible ones for features grid (these should extend into plans-ui data store selectors)
-	const gridPlansForFeaturesGrid = useMemo( () => {
-		return filteredPlansForPlanFeatures.reduce( ( acc, gridPlan ) => {
-			if ( gridPlan.isVisible ) {
-				return [
-					...acc,
-					{
-						...gridPlan,
-						features: planFeaturesForFeaturesGrid[ gridPlan.planSlug ],
-					},
-				];
-			}
-			return acc;
-		}, [] as GridPlan[] );
-	}, [ filteredPlansForPlanFeatures, planFeaturesForFeaturesGrid ] );
+	const gridPlansForFeaturesGrid = useGridPlansForFeaturesGrid( {
+		allFeaturesList: FEATURES_LIST,
+		availablePlans: gridPlans || [],
+		gridPlans: filteredPlansForPlanFeatures,
```

Comment:
I also think none of these (`gridPlans` or `availableGridPlans` - very confusing indeed) should have to be passed through as props. Can we make the hooks a little more self-contained (calling `useGridPlans` internally)? Do you see any blockers/drawbacks to doing so?

Reply:
Sounds good. Would you mind if I tackled that in a followup?

I have some draft changes (https://github.com/Automattic/wp-calypso/pull/87478) which focuses on making the new hooks more self-contained, however it's still a work in progress. Perhaps we can ship this as a small step in the right direction?

Reply:
Sounds good